### PR TITLE
Adding support to style attributes in SVG files

### LINKED ISSUE #404: New feature: provide support for style attributes in SVG files
**Context**
Our current [SVG parser](https://github.com/PyFPDF/fpdf2/blob/master/fpdf/svg.py) has [some limitations](https://pyfpdf.github.io/fpdf2/SVG.html#currently-unsupported-notable-svg-features), among which the lack of support for CSS styling.
This means that the following [simple example from MDN](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/style#example) will not be colored when embedded by `fpdf2` in a PDF document:
```svg
<rect width="80"  height="40" x="10" y="10"
      style="fill: skyblue; stroke: cadetblue; stroke-width: 2;"/>
```

**Goal**
Provide support for parsing `style="..."` attribute in SVG files and convert them to [PDF drawing primitives](https://pyfpdf.github.io/fpdf2/Drawing.html).
Some unit tests should be added covering this feature.
Once tests pass for basic SVG files, more complex SVG example files can be found there, under the GNU license: [Ghostscript examples](https://commons.wikimedia.org/wiki/Example_images#Ghostscript_examples).

---

By implementing this feature you, as a benevolent [FLOSS](https://en.wikipedia.org/wiki/Free/libre_and_open-source_software) developper, will provide access to the community of fpdf2 users to a standard and useful PDF functionality.
You will also be added into the [contributors list & map](https://github.com/PyFPDF/fpdf2#contributors-).
Moreover, by working on this feature, you will learn about PDF & SVG file formats and the lifecycle & structure of a popular Python library.

### PR DESCRIPTION
**Pull request by RedShy**
I implemented the support for SVG with CSS styling elements simply by parsing the style attribute and inserting the key-value pairs as attributes of the element and then reusing the existing code. Close #404

I tested it using the SVG files downloadable from [Ghostscript examples](https://commons.wikimedia.org/wiki/Example_images#Ghostscript_examples) and it works! There are some with linear gradient not supported but the others seems to be okay.
I added some tests in test_svg.py to cover the new feature.

I also have added simple tests for the file transitions.py given that was the least covered with the aim of [increase test coverage](https://github.com/PyFPDF/fpdf2/issues/395).

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

- [ ] The PR description or comment contains a picture of a cute animal (not mandatory but encouraged 😉)

### CODE REVIEW

File: fpdf/svg.py [hunk 1]
```diff
@@ -257,9 +257,27 @@ def optional(value, converter=lambda noop: noop):
...
+    try:
+        style = svg_element.attrib["style"]
+    except KeyError:
+        pass
+    else:
+        for element in style.split(";"):
+            if not element:
+                continue
+
+            attr, value = element.split(":")
```
Comment (Lucas-C) [inline fpdf/svg.py:272]: Maybe we should guard the code againt badly formatted `style`?
Like `style="a:"` or `style=":b"`

File: test/svg/parameters.py [hunk 2]
```diff
@@ -758,6 +688,8 @@ def Gs(**kwargs):
     # discovered while investigatin issue 358:
     pytest.param(svgfile("issue_358b.svg"), id="repeated relative move"),
     pytest.param(svgfile("issue_358.svg"), id="arc start & initial point"),  # issue 358
+    pytest.param(svgfile("Ghostscript_colorcircle.svg"), id="ghostscript colorcircle"),
```
Comment (Lucas-C) [inline test/svg/parameters.py:691]: Could you please also add tests in `test/svg/test_svg.py` that use those 2 files to produces PDFs, and check the generated file with `assert_pdf_equal`? _cf._ https://pyfpdf.github.io/fpdf2/Development.html#assert_pdf_equal-writing-new-tests

**Comment by Lucas-C** [comment]
Note that currently the GitHub Actions pipeline is failing because your code is not formatted with `black`:
https://pyfpdf.github.io/fpdf2/Development.html#code-auto-formatting

Just run `black .` in the repository root directory to fix this.

**Comment by Lucas-C** [comment]
@allcontributors please add @RedShy for code

File: test/svg/parameters.py (hunk 2 above)
Comment (RedShy) [reply test/svg/parameters.py:691]: Actually at first I added these tests:
```
class TestSVGStyleParsing:
    @pytest.mark.parametrize("svg_file", parameters.test_svg_sources[18:20])
    def test_draw_svg_style_to_page(self, tmp_path, svg_file):
        svg = fpdf.svg.SVGObject.from_file(svg_file)

        pdf = fpdf.FPDF(unit="pt", format=(svg.width, svg.height))
        pdf.add_page()
        svg.draw_to_page(pdf)

        assert_pdf_equal(pdf, GENERATED_PDF_DIR / f"{svg_file.stem}.pdf", tmp_path)
```
 but after running them I noticed that they were already executed with:
```
    @pytest.mark.parametrize("svg_file", parameters.test_svg_sources)
    def test_svg_conversion(self, tmp_path, svg_file):
        svg = fpdf.svg.SVGObject.from_file(svg_file)
```
So I removed them

Comment (Lucas-C) [reply test/svg/parameters.py:761]: Oh you are totally right!
That's fine as it is then :)

File: fpdf/svg.py (hunk 1 above)
Comment (RedShy) [reply fpdf/svg.py:272]: Sure! absolutely right. I'm not familiar with pull requests, is it common to put the code here or is it better to just do another commit?
I changed the code with this, now it should be more robust:
```
  pair = element.split(":")
  if len(pair) == 2 and pair[0] and pair[1]:
      attr, value = pair

      svg_element.attrib[attr.strip()] = value.strip() 
```

Comment (Lucas-C) [reply fpdf/svg.py]: > is it common to put the code here or is it better to just do another commit?

I'm not sure to correctly understand your question,
but usually people add commits as part of the same PR to address code reviews feedback.

**Comment by RedShy** [comment]
Thank you! it's very rewarding having someone that is supportive and reviews so fast on the other hand!

**Comment by Lucas-C** [comment]
This feature has been included in the new [v2.5.5](https://github.com/PyFPDF/fpdf2/releases/tag/2.5.5) release!

**Pull request merged**